src/tests/suite/daos_pool.c

```diff
@@ -1005,6 +1006,68 @@ list_containers_test(void **state)
 	print_message("success\n");
 }
 
+static void
+expect_pool_connect_access(test_arg_t *arg0, uint64_t perms,
+			   uint64_t flags, int exp_result)
+{
+	test_arg_t	*arg = NULL;
+	daos_prop_t	*prop;
+	int		 rc;
+
+	rc = test_setup((void **)&arg, SETUP_EQ, arg0->multi_rank,
+			DEFAULT_POOL_SIZE, NULL);
+	assert_int_equal(rc, 0);
+
+	prop = get_daos_prop_with_owner_acl_perms(perms,
+						  DAOS_PROP_PO_ACL);
+
+	while (!rc && arg->setup_state != SETUP_POOL_CREATE)
+		rc = test_setup_next_step((void **)&arg, NULL, prop, NULL);
+	assert_int_equal(rc, 0);
+
+	MPI_Barrier(MPI_COMM_WORLD);
+
+	rc = daos_pool_connect(arg->pool.pool_uuid, arg->group,
+			       &arg->pool.svc, flags,
+			       &arg->pool.poh, &arg->pool.pool_info,
```

Comment:
unsure - should decide if this is fine or if a pattern like the code in test_setup_pool_connect() is necessary here for multi-rank test runs.

Reply:
Since I'm uncertain as well, I decided to rely on the existing test setup infrastructure and updating the test_args_t struct accordingly.